src/js/dp/search.js

```diff
@@ -34,13 +45,28 @@ if (searchContainer) {
       );
     }
 
+    // if it takes more than 500ms to retreive results, show a loading message
+    var loadingPanel = setTimeout(() => {
+      document.querySelector('#results-loading').classList.remove('hide')
+      if (scrollToTop) scrollToTopOfSearch();
+    }, 500);
+
     const responseText = await fetchHtml(`/search${theStringParams}`);
-    if (responseText) {
+
+    if (scrollToTop) scrollToTopOfSearch();
+
+    if (!responseText) {
+      const pTag = document.querySelector('#results-loading p');
+      pTag.innerText = pTag.dataset.errorMessage;
+    } else {
+      // cancel a pending loading message and hide it if it's already showing
+      clearTimeout(loadingPanel);
+      document.querySelector('#results-loading').classList.remove('hide')
```

Comment:
And another iffy; you could probably use the one above (if it's set outside of the function block)